src/lib/storage/proposal/partition_creator.rb

```diff
@@ -211,17 +217,27 @@ def create_non_lvm_complex
 
         # Creates a partition and the corresponding filesystem for each volume
         #
+        # First it tries to create the partitions for the volumes that need to
+        # be in a particular disk.
+        # @raise an error if that's not possible.
+        #
         # It tries to honor the value of #max_start_offset for each volume, but
         # it does not raise an exception if that particular requirement is
         # impossible to fulfill, since it's usually more a recommendation than a
         # hard limit.
         #
         # @param volumes [Array<PlannedVolume>]
         def create_volumes_partitions(volumes)
-          volumes.sort_by_attr(:max_start_offset).each do |vol|
+          volumes.sort_by_attr(:disk, :max_start_offset).each do |vol|
```

Comment:
I don't think currently `volumes` is intended to represent any kind of natural order. But nothing stops us from doing `sort_by_attr` more deterministic than Ruby's `sort` if we feel more comfortable. I will change it to always preserve the original order if all the specified criteria are the same.